__tests__/app.test.js

```diff
@@ -0,0 +1,71 @@
+const request = require("supertest");
+const app = require("../app");
+const db = require("../db/connection"); 
+const seed = require("../db/seeds/seed");
+const testData = require("../db/data/test-data");
+
+
+
+beforeEach(() => {
+    return seed(testData);
+});
+
+afterAll(() => {
+    db.end(); 
+});
+
+describe("GET /api/topics", () => {
+    
+    test('200: should return an array', () => { 
+        return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({ body }) => {
+            const { topics } = body;
+            expect(Array.isArray(topics)).toBe(true);
+            });
+        });
+
+    test('200: should return an array with 3 topic objects', () => { 
+        return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({ body }) => {
+            const { topics } = body;
+            expect(topics.length).toBe(3);
+            });
+        });
+
+    test("200: should respond with an array with all topic objects, each of which will have slug and description properties", () => {
+        return request(app)
+            .get("/api/topics")
+            .expect(200)
+            .then(({ body }) => {
+                const { topics } = body;
+                topics.forEach((el) => {
+                    expect(typeof el).toBe('object');
+                    expect(el).toHaveProperty("slug"), expect.any(String);
+                    expect(el).toHaveProperty("description"), expect.any(String);
+                });
+            });
+    })
+
```

Comment:
I think some of these 200 tests could be combined, nothing wrong with splitting them out though if that's what you prefer!

Reply:
initially i had it all in one test for the 200  but wasnt sure if this was a good approach 
will this suffice 
`
test("200: should respond with an array with 3 topic objects, each of which will have slug and description properties", () => {
        return request(app)
            .get("/api/topics")
            .expect(200)
            .then(({ body }) => {
                const { topics } = body;
                expect(Array.isArray(topics)).toBe(true)
                expect(topics.length).toBe(3)
                topics.forEach((topic) => {
                    expect(typeof topic).toBe('object');
                    expect(topic).toHaveProperty("slug"), expect.any(String);
                    expect(topic).toHaveProperty("description"), expect.any(String);
                });
            });
    })
`

Reply:
![image](https://github.com/i786m/nc-news/assets/71926855/048eb328-12e4-4f93-95aa-e2f87df8144e)
 screenshot of code above for readability